cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,544 @@
+	"k8s.io/kubernetes/pkg/util/tpm"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	TaintKey       string = "Untrusted"
+	LogState       string = "tpm.coreos.com/logstate"
+	UntrustedSince string = "tpm.coreos.com/untrustedsince"
+	TrustedSince   string = "tpm.coreos.com/trustedsince"
+	ValidationTime string = "tpm.coreos.com/validationtime"
+	ConfigName     string = "tpm-manager.coreos.com"
+)
+
+// Flag a node as untrusted
+func invalidateNode(node *api.Node) error {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+	newTaints := []api.Taint{}
+	untrustedTaint := api.Taint{
+		Key:    TaintKey,
+		Value:  "true",
+		Effect: api.TaintEffectNoSchedule,
+	}
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	if err != nil {
+		return fmt.Errorf("Unable to obtain node annotations: %v", err)
+	}
+	for _, taint := range taints {
+		if taint.Key == TaintKey {
+			continue
+		}
+		newTaints = append(newTaints, taint)
+	}
+	newTaints = append(newTaints, untrustedTaint)
+	jsonContent, err := json.Marshal(newTaints)
+	if err == nil {
+		node.Annotations[api.TaintsAnnotationKey] = string(jsonContent)
+	}
+	return err
+}
+
+func trustNode(node *api.Node) error {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+	newTaints := []api.Taint{}
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	if err != nil {
+		return fmt.Errorf("Unable to obtain node annotations: %v", err)
+	}
+	for _, taint := range taints {
+		if taint.Key == TaintKey {
+			continue
+		}
+		newTaints = append(newTaints, taint)
+	}
+	jsonContent, err := json.Marshal(newTaints)
+	if err == nil {
+		node.Annotations[api.TaintsAnnotationKey] = string(jsonContent)
+	}
+	return err
+}
+
+func isTrusted(node *api.Node) bool {
+	if node.ObjectMeta.Annotations == nil {
+		return true
+	}
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	if err != nil {
+		return false
+	}
+	for _, taint := range taints {
+		if taint.Key == TaintKey {
+			return false
+		}
+	}
+	return true
+}
+
+func loadPolicy() (err error) {
+	manager.pcrConfigs = make([]map[string]tpm.PCRConfig, 0)
+
+	if manager.pcrConfig != "" {
+		pcrdata := make(map[string]tpm.PCRConfig)
+		pcrConfig, err := ioutil.ReadFile(manager.pcrConfig)
+		if err != nil {
+			return fmt.Errorf("Unable to read valid PCR configuration %s: %v", manager.pcrConfig, err)
+		}
+		err = json.Unmarshal(pcrConfig, &pcrdata)
+		if err != nil {
+			return fmt.Errorf("Unable to parse valid PCR configuration %s: %v", manager.pcrConfig, err)
+		}
+		for pcrKey, pcrVal := range pcrdata {
+			pcrtmp := pcrVal
+			pcrtmp.Source = manager.pcrConfig
+			pcrdata[pcrKey] = pcrtmp
+		}
+		manager.pcrConfigs = append(manager.pcrConfigs, pcrdata)
+	} else if manager.pcrConfigDir != "" {
+		err = filepath.Walk(manager.pcrConfigDir, func(path string, f os.FileInfo, err error) error {
```

Comment:
I believe we should be checking the incoming err from the the walkFunc. If it cannot walk to file/dir named by the path, then err will be populated (and func can decide how to proceed): https://golang.org/pkg/path/filepath/#WalkFunc